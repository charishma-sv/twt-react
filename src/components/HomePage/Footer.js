import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class Footer extends Component {
  constructor() {
    super();
    this.state = {};
  }
  render() {
    return (
      <div className="mt-3 mx-auto">
        <div
          className=" text-center p-1"
          style={{ fontSize: '13px', overflow: 'hidden' }}
        >
          <Link to="" className="mr-3" style={{ color: 'rgb(91, 112, 131)' }}>
            About
          </Link>
          <Link to="" className="mr-3" style={{ color: 'rgb(91,112,131)' }}>
            Help Center
          </Link>
          <Link to="" className="mr-3" style={{ color: 'rgb(91,112,131)' }}>
            Terms of Service
          </Link>
          <Link to="" className="mr-3" style={{ color: 'rgb(91,112,131)' }}>
            Privacy policy
          </Link>
          <Link to="" className="mr-3" style={{ color: 'rgb(91,112,131)' }}>
            Cookie policy
          </Link>
          <Link to="" className="mr-3" style={{ color: 'rgb(91,112,131)' }}>
            Ads info
          </Link>
          <Link to="" className="mr-3" style={{ color: 'rgb(91,112,131)' }}>
            Blog
          </Link>
          <Link to="" className="mr-3" style={{ color: 'rgb(91,112,131)' }}>
            Status
          </Link>
          <Link to="" className="mr-3" style={{ color: 'rgb(91,112,131)' }}>
            careers
          </Link>
          <Link to="" className="mr-3" style={{ color: 'rgb(91,112,131)' }}>
            Brand Resources
          </Link>
          <Link to="" className="mr-3" style={{ color: 'rgb(91,112,131)' }}>
            Advertising
          </Link>
          <Link to="" className="mr-3" style={{ color: 'rgb(91,112,131)' }}>
            Marketing
          </Link>
          <Link to="" className="mr-3" style={{ color: 'rgb(91,112,131)' }}>
            Twitter for Business
          </Link>
          <Link to="" className="mr-3" style={{ color: 'rgb(91,112,131)' }}>
            Developers
          </Link>
          <Link to="" className="mr-3" style={{ color: 'rgb(91,112,131)' }}>
            Directory
          </Link>
          <Link to="" className="mr-3" style={{ color: 'rgb(91,112,131)' }}>
            Settings
          </Link>
          <div className="text-center  mt-2">
            <Link to="" className="mr-3" style={{ color: 'rgb(91,112,131)' }}>
              &#169; 2021 Twitter, Inc.
            </Link>
          </div>
        </div>
      </div>
    );
  }
}

export default Footer;
