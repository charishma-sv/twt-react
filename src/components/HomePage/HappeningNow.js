import React, { Component } from 'react';
import { Container } from 'react-bootstrap';
import { Link } from 'react-router-dom';

class HappeningNow extends Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    return (
      <Container className="p-3 ">
        <svg
          viewBox="0 0 24 24"
          fill="#1DA1F2"
          className="position-relative d-block"
          style={{ height: '45px', width: '45px' }}
        >
          <g>
            <path d="M23.643 4.937c-.835.37-1.732.62-2.675.733.962-.576 1.7-1.49 2.048-2.578-.9.534-1.897.922-2.958 1.13-.85-.904-2.06-1.47-3.4-1.47-2.572 0-4.658 2.086-4.658 4.66 0 .364.042.718.12 1.06-3.873-.195-7.304-2.05-9.602-4.868-.4.69-.63 1.49-.63 2.342 0 1.616.823 3.043 2.072 3.878-.764-.025-1.482-.234-2.11-.583v.06c0 2.257 1.605 4.14 3.737 4.568-.392.106-.803.162-1.227.162-.3 0-.593-.028-.877-.082.593 1.85 2.313 3.198 4.352 3.234-1.595 1.25-3.604 1.995-5.786 1.995-.376 0-.747-.022-1.112-.065 2.062 1.323 4.51 2.093 7.14 2.093 8.57 0 13.255-7.098 13.255-13.254 0-.2-.005-.402-.014-.602.91-.658 1.7-1.477 2.323-2.41z"></path>
          </g>
        </svg>
        <Container className=" position-relative mt-5 mb-5">
          <span className="text-center h1 display-3 font-weight-bold">
            Happening Now-Lets see change
          </span>
        </Container>
        <Container className=" mb-5">
          <span className="text-center h2 font-weight-bold">
            Join Twitter today
          </span>
        </Container>
        <Container>
          <Link
            to="/signup"
            className="btn btn-custom-color btn-lg w-75 mb-3 rounded-pill buttons "
          >
            <div className="container-fluid h-100 d-flex justify-content-center align-items-center">
              <span>Sign up</span>
            </div>
          </Link>

          <Link
            to="/login"
            className="btn btn-outline-custom-color btn-lg w-75 mb-3 rounded-pill buttons "
          >
            <div className="container-fluid h-100 d-flex justify-content-center align-items-center">
              <span>Log in</span>
            </div>
          </Link>
        </Container>
      </Container>
    );
  }
}

export default HappeningNow;
