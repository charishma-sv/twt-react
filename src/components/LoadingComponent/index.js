import React from 'react';
import { Container, Spinner } from 'react-bootstrap';

const LoadingComponent = () => {
  return (
    <Container className=" vh-100 d-flex justify-content-center align-items-center">
      <Spinner animation="grow" />
    </Container>
  );
};

export default LoadingComponent;
