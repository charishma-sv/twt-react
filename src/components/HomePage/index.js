import React, { useContext } from 'react';
import { Container, Row, Col } from 'react-bootstrap';
import Footer from './Footer';
import HappeningNow from './HappeningNow';
import LogoBackground from './LogoBackground';

import '../../styles/scss/main.scss';
import { UserContext } from '../../providers/UserProvider';

const HomePage = () => {
  const user = useContext(UserContext);

  return (
    <Container fluid className="p-0">
      <Row className="no-gutters">
        <Col
          sm={12}
          lg={8}
          xl={5}
          className="order-lg-last d-flex align-items-center justify-content-center"
        >
          <HappeningNow />
        </Col>
        <Col sm={12} lg={4} xl={7} className="order-lg-first ">
          <LogoBackground />
        </Col>
      </Row>
      <Row className="m-3">
        <Footer />
      </Row>
    </Container>
  );
};

export default HomePage;
