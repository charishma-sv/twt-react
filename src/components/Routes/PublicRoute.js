import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import LoadingComponent from '../LoadingComponent';

const PublicRoute = ({
  component: Component,
  user,
  authListner,
  updateUser,
  isLoggedIn,
  ...rest
}) => {
  return (
    // <Route
    //   {...rest}
    //   render={(props) => {
    //     console.log('props in public', props);
    //     if (user) {
    //       return <Redirect to="/feed" />;
    //     } else {
    //       return <Component />;
    //     }
    //   }}
    // />
    <Route {...rest}>
      {user === null ? (
        <LoadingComponent />
      ) : user ? (
        <Redirect to="/feed" />
      ) : (
        <Component updateUser={updateUser} />
      )}
    </Route>
  );
};

export default PublicRoute;
