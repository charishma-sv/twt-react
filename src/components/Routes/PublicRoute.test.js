import React from 'react';
import {
  BrowserRouter,
  MemoryRouter,
  Redirect,
  Router,
} from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import renderer from 'react-test-renderer';

import HomePage from '../HomePage';
import PublicRoute from './PublicRoute';
import LoadingComponent from '../LoadingComponent';

describe('homepage render on not autheticated user', () => {
  it('should render homepage if not user is undefined', async () => {
    const AComponent = () => <HomePage />;

    let user = undefined;
    const path = '/';
    const propts = {
      path: '/',
    };

    const { root: instance } = renderer.create(
      <BrowserRouter>
        <PublicRoute component={AComponent} {...propts} user={user} />
      </BrowserRouter>
    );
    const HomepageComponent = instance.findByType(HomePage);
    expect(HomepageComponent).toBeDefined();
  });
});

describe('loadingComponent snapshot on user is null', () => {
  it('snaphot on user exits', () => {
    const AComponent = () => <LoadingComponent />;

    let user = 'hello';
    const path = '/';
    const propts = {
      path: '/',
    };
    const history = createMemoryHistory({ initialEntries: ['/feed'] });
    const tree = renderer.create(
      <Router history={history}>
        <PublicRoute component={AComponent} {...propts} user={user} />
      </Router>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render LoadingComponent if user is null', async () => {
    const AComponent = () => <LoadingComponent />;

    let user = null;
    const path = '/';
    const propts = {
      path: '/',
    };

    const { root: instance } = renderer.create(
      <BrowserRouter>
        <PublicRoute component={AComponent} {...propts} user={user} />
      </BrowserRouter>
    );
    const LoadingComp = instance.findByType(LoadingComponent);
    expect(LoadingComp).toBeDefined();
  });
});

describe('feed render on user exists', () => {
  it('snaphot on user exits', () => {
    const AComponent = () => <Feed />;

    let user = { email: 'john' };
    const path = '/feed';
    const propts = {
      path: '/feed',
    };
    const history = createMemoryHistory({ initialEntries: ['/feed'] });
    const tree = renderer.create(
      <Router history={history}>
        <PublicRoute component={AComponent} {...propts} user={user} />
      </Router>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render Feed if user exists', async () => {
    const AComponent = () => <Feed />;

    let user = 'john';
    const path = '/feed';
    const propts = {
      path: '/feed',
    };
    const history = createMemoryHistory({ initialEntries: ['/feed'] });

    const { root: instance } = renderer.create(
      <Router history={history}>
        <PublicRoute component={AComponent} {...propts} user={user} />
      </Router>
    );

    const RedirectComp = instance.findByType(Redirect);
    expect(RedirectComp).toBeDefined();
  });
});
