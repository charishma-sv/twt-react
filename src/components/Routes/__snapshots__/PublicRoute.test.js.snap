// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feed render on user exists snaphot on user exits 1`] = `null`;

exports[`loadingComponent snapshot on user is null snaphot on user exits 1`] = `null`;
